tiledb/sm/array/array.h

```diff
@@ -778,6 +778,46 @@ class Array {
    */
   void non_empty_domain(void* domain, bool* is_empty);
 
+  /**
+   * Returns the non-empty domain of the opened array on the given dimension.
+   * This is the union of the non-empty domains of the array fragments.
+   *
+   * @param field_name The dimension name.
+   * @param domain The domain to be retrieved.
+   * @param is_empty `true` if the non-empty domain (and array) is empty.
+   */
+  void non_empty_domain_from_name(
+      std::string_view field_name, void* domain, bool* is_empty);
```

Comment:
I had not realized we could just straight to using `string_view` in the signature, but it works fine. Very good.